### ISSUE TITLE
Large/very long albums break player

### ISSUE THREAD

**Issue by robbevp** [accentor/android#143]
**Describe the bug**
Albums with a lot of tracks/with a long total length make the player break in several ways.
(I'm not sure whether it is the amount of tracks or the total length of the tracks, i noticed this when playing an album with 310 tracks and - according to my estimations - a total length of 15 - 16hrs)
Some issues i noticed:
* The view takes a long time to update
* Not all tracks are loaded into playlist
* Playtime no longer displays properly
* Album art isn't shown
* If starting another album, view of playlist isn't updated until app is reopened

**To Reproduce**

1. Open app
2. Play album with a lot of tracks
3. Try and open playlist view
4. Compare tracks in playlist to tracks in web

**Screenshots**
![Screenshot_20210207-195833_Accentor](https://user-images.githubusercontent.com/48474670/107156739-6cfa3780-6980-11eb-820a-d23ba10bed6c.jpg)

**Additional context**
Might already be solved by #99 - but seems good to track the issue either way

**Comment by chvp** [comment]
I can't reproduce most of this since #99 was merged. The only issue I noticed when doing some testing is that adding tracks to the current playlist becomes progressively slower the more tracks there are in the playlist (e.g. playing "Route One (24 Hour Version)" and then adding any other album: the tracks of the other album trickle in quite slowly).

**Comment by robbevp** [comment]
It might be good to test this on my device once we have a new release. I think your phone is a lot more performant than mine, so it might be that it's the same issue but just more prominent

**Comment by chvp** [comment]
Probably not, I noticed most of the issues you listed before #99 as well.

**Comment by robbevp** [comment]
If I try to view the playlist with Route One (24 Hour Version), my phone consistently crashes. (I can play the album, as long as I don't open the playlist)

**Comment by chvp** [comment]
Phone, or the app?

**Comment by robbevp** [comment]
phone

**Comment by robbevp** [comment]
This is fixed since v15 for me. The same album the consistently crashed the app when playing and/or scrolling through tracklist now works without issue.
>  The only issue I noticed when doing some testing is that adding tracks to the current playlist becomes progressively slower the more tracks there are in the playlist (e.g. playing "Route One (24 Hour Version)" and then adding any other album: the tracks of the other album trickle in quite slowly).
Even is this now works instantly

**Comment by chvp** [comment]
Ah, hadn't tested yet. Great news! That means this can be closed?

**Comment by chvp** [comment]
Hmm, the route one + another album problem is still a problem for me.

**Comment by chvp** [comment]
I wonder if this is because I'm always on a debug build. I'll check tomorrow.

**Comment by robbevp** [comment]
@chvp have you tried this outside of debug mode? Seems very weird that it would work without issues on my device and not on yours

**Comment by chvp** [comment]
I did, it was not fixed. However, I'm pretty sure I figured out what the problem is here. We load a 500x500 bitmap into every metadata object. These metadata objects gets replicated to every media session listener, of which there are at least two in the app (probably more until #234 is merged). The OS probably has at least one more listener. Every bitmap is 750kB (or even 1MB if there is an alpha channel, not sure about that). This obviously results in quite a bit of memory pressure for large queues.

The reason you didn't have any problems is probably due to #220.
